### ISSUE TITLE
/markets and markets/.../estimated-best-ask-price return inconsistent prices

### ISSUE THREAD

**Issue by fleupold** [gnosis/dex-services#1344]
Exaggerating on structure here since these price issues can easily be overwhelming...

## Issue

`/markets` returns prices in quote, e.g. https://price-estimate-mainnet.dev.gnosisdev.com/api/v1/markets/WETH-DAI shows prices in DAI

> ... {"price":404.67014059206866,"volume":246744853556.0} ...

`markets/.../estimated-best-ask-price` however shows prices in base, e.g. https://price-estimate-mainnet.dev.gnosisdev.com/api/v1/markets/WETH-DAI/estimated-best-ask-price?unit=atoms returns prices in WETH 

> 0.002295201341664352

despite the comment in the code saying otherwise:

https://github.com/gnosis/dex-services/blob/03075b0c419f6a69f6ef6f62e37db37e0f4eb625/price-estimator/src/models.rs#L64-L68

## Why is that?

I believe this comes from the fact that we use the bid pair to estimate the best ask price. This itself is correct as we want to "simulate" a person trying to buy a tiny amount of base from the people offering it to get the "best offer" (best ask):

https://github.com/gnosis/dex-services/blob/03075b0c419f6a69f6ef6f62e37db37e0f4eb625/price-estimator/src/filter.rs#L310-L315

In the snippet below we can see that the returned price is given in buy/sell token and thus in base/quote (in the example WETH per $) rather than quote/base (in the example $ per WETH). The latter is what the orderbook is showing.

https://github.com/gnosis/dex-services/blob/03075b0c419f6a69f6ef6f62e37db37e0f4eb625/pricegraph/src/api/price_estimation.rs#L14-L16

## Proposed Fix

I believe it is fine to just invert the price in the filter, as fees are accounted in a way to make the buy amount smaller (lowering best the best price available in base). Inverting this would lead to increasing the best ask. This makes the spread larger than in a no-fee model which seems reasonable.

@e00E @nlordell could you please double check if my thinking is correct and if it's safe to just invert the price before creating the result type:

https://github.com/gnosis/dex-services/blob/03075b0c419f6a69f6ef6f62e37db37e0f4eb625/price-estimator/src/filter.rs#L315-L318

Note, that I don't think this has anything to do with base-units or atoms, but I noticed this behavior when coding the base unit support for the best ask route.

**Comment by nlordell** [comment]
🙈 best ask price is using the bid token pair 

https://github.com/gnosis/dex-services/blob/03075b0c419f6a69f6ef6f62e37db37e0f4eb625/price-estimator/src/filter.rs#L310

**Comment by fleupold** [comment]
> best ask price is using the bid token pair

That's fine, isn't it? It asks for a price when selling a marginal amount into the ask curve (thus taking the bid token pair).
I actually compared the result with what I see in the orderbook and it makes sense to be the best ask for this market, just that it is the inverted price.

E.g. right now best ask for WETH WBTC (needed a non overlapping book) seems to be ~ $0.047

<img width="1285" alt="Screen Shot 2020-08-19 at 19 14 45" src="https://user-images.githubusercontent.com/1200333/90668427-6223dd00-e250-11ea-8d5d-1bbbc2f016f4.png">


The response of best ask (http://localhost:8080/api/v1/markets/WETH-WBTC/estimated-best-ask-price) is

22.79287489177357 which inverted is **0.04387335975598765**


The inverse market http://localhost:8080/api/v1/markets/WBTC-WETH/estimated-best-ask-price returns the best bid in quote

**0.03510607972374784**

<img width="1289" alt="Screen Shot 2020-08-19 at 19 14 51" src="https://user-images.githubusercontent.com/1200333/90668654-b0d17700-e250-11ea-9b19-df5e8aeff6a1.png">


However, it looks actually as if fees are applied the wrong way, since the estimated spread is tighter than the visual one 🙈

**Comment by e00E** [comment]
> could you please double check if my thinking is correct and if it's safe to just invert the price before creating the result type:

Yes I think you're right. Your explanation makes sense and I re-read the code.
